modules/siddhi-core/src/main/java/io/siddhi/core/util/parser/CollectionExpressionParser.java

```diff
@@ -505,7 +520,7 @@ public static CollectionExecutor buildCollectionExecutor(CollectionExpression co
                                     outputExpectsExpiredEvents, siddhiQueryContext);
                             aCollectionExecutor = buildCollectionExecutor(rightCollectionExpression,
                                     matchingMetaInfoHolder, variableExpressionExecutors, tableMap,
-                                    isFirst, processingMode, outputExpectsExpiredEvents, siddhiQueryContext);
+                                    isFirst, processingMode, outputExpectsExpiredEvents, siddhiQueryContext, false);
```

Comment:
shouldn't this be `isCache`?  or the isCache is only applied at the root level ?

Reply:
the constraints we have for routing a query to cache results in collectionExpression being either AttributeCollectionExpression, CompareCollectionExpression, or AndMultiPrimaryKeyCollectionExpression. So the rest of the cases I have sent false. Please advise if we will have cache constraint satisfaction in other cases also